src/tests/integration/test-data/foundry/test/EmitContractTest.t.sol

```diff
@@ -36,4 +36,11 @@ contract EmitContractTest is Test {
         // The event we get
         emitter.t();
     }
+
+    function testExpectEmitLessTopics() public {
+        ExpectEmit emitter = new ExpectEmit();
+        vm.expectEmit(true, true, true, false);
+        emit Transfer(address(this), address(1337), 1337);
+        emitter.t();
```

Comment:
Doesn't `t()` emit the event with the same number of topics? 
```
    function t() public {
        emit Transfer(msg.sender, address(1337), 1337);
    }
```

Reply:
Yes, it does, but in `vm.expectEmit(true, true, true, false);` we are saying to compare all 4 topics. The `#checkTopics` returned false previously if the topic list was empty but we still wanted to check them.